src/components/Contacts/ContactDetails/ContactDetailsTab/People/Items/EditContactOtherModal/EditContactOtherModal.tsx

```diff
@@ -0,0 +1,304 @@
+    timezone: yup.string().nullable(),
+    website: yup.string().nullable(),
+  });
+
+  const onSubmit = async (attributes: ContactUpdateInput) => {
+    try {
+      await updateContactOther({
+        variables: {
+          accountListId,
+          attributes: {
+            id: attributes.id,
+            churchName: attributes.churchName,
+            preferredContactMethod: attributes.preferredContactMethod,
+            locale: attributes.locale,
+            timezone: attributes.timezone,
+            website: attributes.website,
+          },
+        },
+      });
+      enqueueSnackbar(t('Contact updated successfully'), {
+        variant: 'success',
+      });
+    } catch (error) {
+      enqueueSnackbar(error.message, { variant: 'error' });
+      throw error;
+    }
+  };
+
+  const localizedContactMethod = (method: string) => {
+    switch (method) {
+      case PreferredContactMethodEnum.Sms:
+        return t('SMS');
+      case PreferredContactMethodEnum.PhoneCall:
+        return t('Phone Call');
+      case PreferredContactMethodEnum.Email:
+        return t('Email');
+      case PreferredContactMethodEnum.Facebook:
+        return t('Facebook');
+      case PreferredContactMethodEnum.Instagram:
+        return t('Instagram');
+      case PreferredContactMethodEnum.WeChat:
+        return t('WeChat');
+      case PreferredContactMethodEnum.WhatsApp:
+        return t('WhatsApp');
+    }
+  };
+
+  return (
+    <Modal
+      isOpen={isOpen}
+      title={t('Edit Contact Other Details')}
+      handleClose={handleClose}
+    >
+      <Formik
+        initialValues={{
+          id: contact.id,
+          churchName: contact.churchName,
+          preferredContactMethod: contact.preferredContactMethod,
+          locale: contact.locale,
+          timezone: contact.timezone,
+          website: contact.website,
+        }}
+        validationSchema={contactOtherSchema}
+        onSubmit={onSubmit}
+      >
+        {({
+          values: {
+            churchName,
+            preferredContactMethod,
+            locale,
+            timezone,
+            website,
+          },
+          handleChange,
+          handleSubmit,
+          isSubmitting,
+          isValid,
+        }): ReactElement => (
+          <form onSubmit={handleSubmit}>
+            <DialogContent dividers>
+              <ContactEditContainer>
+                <ContactInputWrapper>
+                  <FormControl fullWidth={true}>
+                    <InputLabel id="preferred-contact-method-select-label">
+                      {t('Preferred Contact Method')}
+                    </InputLabel>
+                    <Select
+                      labelId="preferred-contact-method-select-label"
+                      value={preferredContactMethod}
+                      onChange={handleChange('preferredContactMethod')}
+                      inputProps={{
+                        'aria-label': t('Preferred Contact Method'),
```

Comment:
Do you need this if you're specifying a label?

Reply:
Hmmm, probably not. I double-check and remove it if need be.

Reply:
It's not needed for the selects, because of the form control and label component. But it is needed for the ```TextField``` components.

Reply:
The `label` prop on `TextField`s doesn't accomplish the same thing?

Reply:
No, it doesn't seem to. When I remove it the tests fail to find those elements.

Reply:
Weird...